train.py

```diff
@@ -67,7 +80,7 @@ def train_model(
 ) -> nn.Module:
     # Get the index for padding token
     pad_idx = train_dataset.src_encoder.token_to_id["<blank>"]
-    src_vocab_size, tgt_vocab_size = vocab_sizes(cfg)
+    src_vocab_size, tgt_vocab_size = train_dataset.src_encoder.vocab_size, train_dataset.tgt_encoder.vocab_size
```

Comment:
this makes the code more versatile

Reply:
In my opinion having a separate way to calculate those values is better - you don't need to create a dataset to know those numbers, and the same line of code can be used in training, validation, dashboards and any other place where no "train_dataset" exist.

The design idea here is to have a function that acts as a "lens" to a problem - it's a term commonly used in functional programming, although it's much more complicated there :)

Reply:
okay :)